### PR TITLE
configure openshift prometheus for hub's alertmanager.

### PR DESCRIPTION
**Pull request by morvencao**
ref: open-cluster-management/backlog#12110
resolve: https://github.com/open-cluster-management/backlog/issues/12477 and https://github.com/open-cluster-management/backlog/issues/12474
depends on: https://github.com/open-cluster-management/multicluster-observability-operator/pull/483

Signed-off-by: morvencao <lcao@redhat.com>

### CODE REVIEW

File: controllers/observabilityendpoint/ocp_monitoring_config_test.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	err = c.Get(ctx, types.NamespacedName{Name: hubAmRouterCASecretName,
+		Namespace: promNamespace}, foundHubAmRouterCASecret)
+	if err == nil || !errors.IsNotFound(err) {
+		t.Fatalf("the secret %s should be deleted", hubAmRouterCASecretName)
+	}
+
+	err = revertClusterMonitoringConfig(ctx, c)
+	if err != nil {
+		t.Fatalf("Run into error when try to revert cluster-monitoring-config configmap twice: (%v)", err)
+	}
```
Comment (clyang82) [inline controllers/observabilityendpoint/ocp_monitoring_config_test.go:184]: You may add more test cases to cover:
1. has `cluster-monitoring-config` w/o prometheusK8s
2. has `cluster-monitoring-config` w/ prometheusK8s
3. has `cluster-monitoring-config` w/o additionalAlertManagerConfigs
4. 3. has `cluster-monitoring-config` w additionalAlertManagerConfigs

File: README.md [hunk 2]
```diff
@@ -67,15 +67,20 @@ metadata:
   name: hub-info-secret
 type: Opaque
 data:
+    clusterName: ***
```
[the remark this answers is not in the document]
Comment (morvencao) [reply README.md:70]: I remember cluster name was move out of hub-info.yaml 
see: https://github.com/open-cluster-management/multicluster-observability-operator/pull/535/files#diff-734c34b115a4b79ffd95cdbd2bffcee860661ad0fc34875085f4a347492430f6
https://github.com/open-cluster-management/multicluster-observability-operator/blob/main/controllers/placementrule/manifestwork.go#L248-L249

File: controllers/observabilityendpoint/metrics_collector.go [hunk 3]
```diff
@@ -64,8 +64,10 @@ type Rule struct {
 
 // HubInfo is the struct for hub info
 type HubInfo struct {
-	ClusterName string `yaml:"cluster-name"`
-	Endpoint    string `yaml:"endpoint"`
+	ClusterName             string `yaml:"cluster-name"`
+	Endpoint                string `yaml:"endpoint"`
+	HubAlertmanagerEndpoint string `yaml:"hub-alertmanager-endpoint"`
+	HubAlertmanagerRouterCA string `yaml:"hub-alertmanager-router-ca"`
```
[the remark this answers is not in the document]
Comment (morvencao) [reply controllers/observabilityendpoint/metrics_collector.go:70]: I think we need to update the struct in MCO and use that type as source of truth.
Will raise PR for that.

[the remark this answers is not in the document]
Comment (morvencao) [reply controllers/observabilityendpoint/metrics_collector.go:70]: I raise PR against MCO, see: https://github.com/open-cluster-management/multicluster-observability-operator/pull/579
We should use that struct definition and should not redefine in this repo.

File: controllers/observabilityendpoint/ocp_monitoring_config.go [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+	cmomanifests "github.com/openshift/cluster-monitoring-operator/pkg/manifests"
+)
+
+const (
+	hubAmAccessorSecretName            = "observability-alertmanager-accessor"
+	hubAmAccessorSecretKey             = "token"
+	hubAmRouterCASecretName            = "hub-alertmanager-router-ca"
+	hubAmRouterCASecretKey             = "service-ca.crt"
+	additionalPromToAmConfigSecretName = "prometheus-ocm-am-config"
+	additionalPromToAmConfigKey        = "additional-alertmanager-configs.yaml"
```
[the remark this answers is not in the document]
Comment (morvencao) [reply controllers/observabilityendpoint/ocp_monitoring_config.go:28]: good catch!

File: controllers/observabilityendpoint/ocp_monitoring_config.go [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+	if err != nil {
+		log.Error(err, "error deleting the hub-alertmanager-router-ca secret")
+		return err
+	}
+	log.Info("the hub-alertmanager-router-ca secret is deleted")
+	return nil
+}
+
+// createHubAmAccessorTokenSecret creates the secret that contains access token of the Hub's Alertmanager
+func createHubAmAccessorTokenSecret(ctx context.Context, client client.Client) error {
```
[the remark this answers is not in the document]
Comment (morvencao) [reply controllers/observabilityendpoint/ocp_monitoring_config.go:102]: main concern is that the two secrets are too much different, including secret data key, name...

File: controllers/observabilityendpoint/ocp_monitoring_config_test.go (hunk 1 above)
Comment (morvencao) [reply controllers/observabilityendpoint/ocp_monitoring_config_test.go:248]: thanks, added more cases for `cluster-monitoring-config` reconcile process.

File: controllers/observabilityendpoint/ocp_monitoring_config.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (morvencao) [reply controllers/observabilityendpoint/ocp_monitoring_config.go]: done.

**Pull request merged**